main_test.go

```diff
@@ -122,6 +127,62 @@ func binsubstr(ts *testscript.TestScript, neg bool, args []string) {
 	}
 }
 
+func binsubint(ts *testscript.TestScript, neg bool, args []string) {
+	if len(args) < 2 {
+		ts.Fatalf("usage: binsubint file subint...")
+	}
+	data := ts.ReadFile(args[0])
+	var failed []string
+	for _, subIntStr := range args[1:] {
+		subInt, err := strconv.Atoi(subIntStr)
+		if err != nil {
+			ts.Fatalf("%v", err)
+		}
+
+		b := make([]byte, 8)
+		binary.LittleEndian.PutUint64(b, uint64(subInt))
```

Comment:
Seems unnecessary as it is already checked before, that the numbers appear as little endian in the  normal binary.

Reply:
I don't follow; what I'm saying is that this check won't be useful if the current architecture is big-endian instead of little-endian, so we should check both. Or, if you want to be extra precise, use the right endianness for the current system.